### PR TITLE
Fix ingrain for Phantump and Trevenant (#856)

### PR DESCRIPTION
**Pull request by charliegiang**
Fixes issue https://github.com/pagefaultgames/pokerogue/issues/856

Before fix:
https://github.com/pagefaultgames/pokerogue/assets/146129103/f4dccd4b-1cce-4466-918f-ff30fd1d3fcb

After fix: 
https://github.com/pagefaultgames/pokerogue/assets/146129103/36c75dd2-6039-4cbb-be57-3f75f4d6be6d

### CODE REVIEW

**Comment by Brandon-Rodgers** [comment]
This fixes one part of not getting the heal effects, but then adds a bug where those two Pokemon cannot switch out now. Shouldn't be swapping one bug for another.

**Comment by Brandon-Rodgers** [comment]
https://github.com/pagefaultgames/pokerogue/assets/9848614/378c1128-661f-4fd2-8a73-7c2640bcae0d

Now you cannot switch out Trevenant or Phantump.

[Ingrain](https://bulbapedia.bulbagarden.net/wiki/Ingrain_(move))
>[Ghost-type](https://bulbapedia.bulbagarden.net/wiki/Ghost_(type)) Pokémon under the effects of Ingrain are now able to switch out. However, they still cannot be forced to switch by moves that switch the target out.

File: src/data/battler-tags.ts [hunk 1]
```diff
@@ -117,7 +118,11 @@ export class TrappedTag extends BattlerTag {
   }
   
   canAdd(pokemon: Pokemon): boolean {
-    return !pokemon.isOfType(Type.GHOST) && !pokemon.getTag(BattlerTagType.TRAPPED);
+    const isGhost = pokemon.isOfType(Type.GHOST);
+    const isTrapped = pokemon.getTag(BattlerTagType.TRAPPED);
+    const isAllowedGhostType = pokemon.species.speciesId === Species.PHANTUMP || pokemon.species.speciesId === Species.TREVENANT;
+
+    return (!isGhost && !isTrapped) || (isAllowedGhostType && !isTrapped);
```
Comment (danieland) [inline src/data/battler-tags.ts:125]: This looks like it can be simplified to:
`return !isTrapped && (!isGhost || isAllowedGhostType);`

**Comment by charliegiang** [comment]
> 2024-05-16.18-53-46.mp4
> 
> Now you cannot switch out Trevenant or Phantump.
> 
> [Ingrain](https://bulbapedia.bulbagarden.net/wiki/Ingrain_(move))
> 
> > [Ghost-type](https://bulbapedia.bulbagarden.net/wiki/Ghost_(type)) Pokémon under the effects of Ingrain are now able to switch out. However, they still cannot be forced to switch by moves that switch the target out.

Gotcha, I didn't know that was the most recent behavior. Thanks for checking. I'll look into fixing this 👍

File: src/phases.ts [hunk 2]
```diff
@@ -1816,7 +1817,14 @@ export class CommandPhase extends FieldPhase {
...
               this.scene.currentBattle.turnCommands[this.fieldIndex - 1].skip = true;
           } else if (trapTag) {
-            if (!isSwitch) {
+            if(trapTag.sourceMove === Moves.INGRAIN && this.scene.getPokemonById(trapTag.sourceId).isOfType(Type.GHOST)) {
+              success = true;
+              this.scene.currentBattle.turnCommands[this.fieldIndex] = isSwitch 
+              ? { command: Command.POKEMON, cursor: cursor, args: args }
+              : { command: Command.RUN };
+              break;
+            }
```
Comment (charliegiang) [inline src/phases.ts:1826]: Allowing switch for ingrained ghost types cc: @Brandon-Rodgers

**Review by Madmadness65 — APPROVED** [review]

**Pull request merged**